### DISCUSSION TITLE
Piper-phonemize for Windows?

### DISCUSSION THREAD

**Discussion by JohnyB7** [rhasspy/piper#173] [Q&A]
I'd love to try out Piper, but I'm not sure how to install it on Windows 10 as the pip install requires piper-phonemize and I'm not sure how to get it running on Win10.

**Comment by StoryHack** [comment]
If you have a relatively modern machine you can use windows subsystem for linux (wsl) to install and run piper. It basically makes a virtual linux machine for you. You do need to learn a bit about using wsl, though, so it's not so straightforward and just having a windows distribution.

**Comment by JohnyB7** [comment]
Thanks you for the explanation, but sadly this is far too complicated for me and I don't have enough free time to learn how to do it.
I will simply use Tortoise TTS instead of Piper.
I don't know why the developer decided to make the Piper project dependent on something that isn't compatible with 95% of the OS market and then abandon the project.
I hope someone forks the project and makes it more accessible to the vast majority of potential users.

**Reply by StoryHack** [reply]
I think the dev(s) is mostly shooting at tiny devices that run linux, not desktop users. Things like Amazon's echo device and raspberry pis.

**Reply by JohnyB7** [reply]
Fair enough. I guess I've misunderstood the description. He should replace "optimized for raspberry" with "a raspberry pi exclusive project". If I saw a drinking bottle that is "optimized for outdoor activity" I would not assume that it is dysfunctional indoors. My mistake.
I still hope that he decides to make it compatible with Windows and Mac somedays.

**Reply by Feanix-Fyre** [reply]
I tried working on a Windows port with my limited aptitude. I was able to figure out, I think, that piper phonemizer is primarily just espeak-ng with one single extra function. Espeak has windows options. I just can't figure out how to port the function over. I don't really want to have to have WSL for, in my experience, the best real time TTS model currently out there combined with better voice dataset collection instructions than "As muCh DaTA as PossIBlE!!!!1!1but hi quaLItY samPles stolen....I mean BORROWED from YouTube/anime/tv shows/movies/etc."

From what I've seen in other AI dev for 2023, if this had a gradio GUI and a Jupyter Notebook specifically for Google Colab, it would be on the radar of some VC investors. 

I don't really care about that thought. I'm trying to port it into an android app so that when I'm having a mental health episode, I can still communicate. But I want that android app to either be able to do on-device or off-device inference, and I want off-device handled by my home windows PC used for multiple Windows specific applications, while running the lightweight and high quality piper. 

But I can't. Cause of a phonemizer and my lack of aptitude. 

Alas.

**Reply by synesthesiam** [reply]
Work is being done on a Windows port. I'm the only dev, and I work in Linux so that has been the target platform.

Windows is a pain to support because I can't just pull a Docker build image and create an automated workflow. I have to use a VM or GitHub actions since cross-compilation hasn't worked either.

**Reply by Feanix-Fyre** [reply]
I tried tackling the GitHub Actions compilation from where you got stuck on piper-phonemizer. You can check my fork for how far I got. One issue was that windows uses slashes in the opposite direction from Linux so I swapped those out. Another issue was that it couldn't find something so I told it where it was directly. And then I got stuck cause I didn't actually know what I was doing.

**Reply by synesthesiam** [reply]
A Windows pre-release is now available for testing: https://github.com/rhasspy/piper/releases/tag/2023.9.9-1

**Reply by Feanix-Fyre** [reply]
I had an issue with json as an input, but it does perform its MVP (double entendre) for what it claims: lightweight, faster than real time, easy to use, easy to train, high quality TTS model. I just call it with a batch or python script if I want to do a bunch of inference. I want to recompare to see if there is any difference in how it is breaking up long sentences and how the colab/linux version would but not necessarily within the scope of the project or your intentions as a developer. My evaluation is that its intention is to be for the edge so this windows port is just a blessing and I'll take it as it is. With how lightweight and fast it is, it's not a problem for it to be called over and over. Not necessarily the most efficient but I was working with Tortoise before this so....... :) I'm a happy camper. Thank you, kindly, for the hard work.

**Reply by Kudos6505** [reply]
Did you get the json or audio streaming to work? I worked around the json input by using a txt file and "type" command. I still don't know how to stream audio, as aplay isn't on windows.

**Comment by muhammad-meraj** [comment]
I am trying to use piper in the project which runs on arm-wince device. If there is any compiled version or binary for arm-wince device? 
Or any guide to compile it for arm-wince device. Thanks

**Reply by csukuangfj** [reply]
https://github.com/k2-fsa/sherpa-onnx
uses piper-phonemize and it supports Linux, windows, macOS, iOS, Android.

**Comment by sam-14uel** [comment]
> I'd love to try out Piper, but I'm not sure how to install it on Windows 10 as the pip install requires piper-phonemize and I'm not sure how to get it running on Win10.

Same thing with me
